tests/adversarial/test_adversarial_manifest_quality.py

```
@@ -75,6 +81,115 @@ def _write_manifest_without_schema(path: Path, controls: dict, status: str) -> N
+        {"metrics": {"near_misses": 1}},
+    ]
+
+
+def test_collect_paths_reports_bad_inputs_and_discovers_yaml(tmp_path: Path) -> None:
+    """Path collection should fail clearly for bad inputs and recurse over YAML manifests."""
+    with pytest.raises(FileNotFoundError, match="Manifest input missing"):
+        _collect_paths([tmp_path / "missing"])
+
+    text_file = tmp_path / "not_manifest.txt"
+    text_file.write_text("not yaml", encoding="utf-8")
+    with pytest.raises(ValueError, match="must use .yml or .yaml"):
```

Ruff (0.15.17)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)